### ISSUE TITLE
Model collapses to single-class predictions with ~10% positive rate dataset

### ISSUE THREAD

**Issue by AbhinavKhareTech** [kumo-ai/kumo-rfm#69]
### Summary

When running KumoRFM on a BNPL/trade credit dataset with ~10-12% positive labels (buyer defaults), the model frequently collapses to all-zero predictions. The training logs indicate that internal sampling drops all positive examples. 

### Details

This occurs even though the full dataset has a clear positive class representation and labels are correctly set before masking the test split.

### Steps to Reproduce

1. Build a five-table relational graph (buyers, suppliers, invoices, payments, credit_lines) with ~400 buyers, ~45 labeled as defaulters (~11% positive rate)
2. Mask test buyer labels via `pd.NA` for missing-value imputation
3. Set `is_defaulter` as the target column with `stype = "numerical"`
4. Run prediction:

```python
query = "PREDICT buyers.is_defaulter = 1 FOR buyers.buyer_id IN ('B0001', 'B0002', ...)"
result = model.predict(query)
```

5. All returned predictions are 0 (or constant), regardless of whether the buyer is a known defaulter in the held-out ground truth

### Expected Behavior

With ~11% positive rate in the training portion of the graph, the model should produce meaningful probability separation between positive and negative examples. At minimum, the system should warn if its internal sampling has produced a single-class training set.

### Observed Behavior

- Predictions are constant (all zeros or near-zero)
- No warning or error is raised
- The training log line `0.00% positive cases` suggests the internal context sampling is not preserving label distribution

### Suggested Fix

- Enforce stratified sampling or minimum positive representation during internal context selection
- Raise an explicit warning or error if sampled training data contains only one class (e.g., "Warning: context contains 0 positive examples. Predictions may be degenerate.")
- Surface class distribution at each stage (global dataset, sampled context, predictions) in logs

### Environment

- kumoai SDK (latest pip install)
- Python 3.10
- Google Colab / Jupyter

### Contact (optional)

khare.abhinav@gmail.com
